server/src/main/kotlin/com/kamelia/hedera/core/auth/Session.kt

```diff
@@ -0,0 +1,50 @@
+package com.kamelia.hedera.core.auth
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import com.kamelia.hedera.util.Environment
+import java.util.*
+import kotlinx.serialization.Serializable
+
+const val USER_ID_CLAIM = "userId"
+const val SESSION_ID_CLAIM = "sessionId"
+
+@Serializable
+data class TokenContainer(val token: String, val expiration: Long)
+
+@Serializable
+class Session(
+    val accessToken: TokenContainer,
+    val refreshToken: TokenContainer,
+    var lastUsed: Long,
+) {
+    companion object {
+        fun from(userId: UUID, sessionId: UUID): Session {
+            val now = System.currentTimeMillis()
+
+            val accessTokenExpiration = now + Environment.expirationAccess
+            val accessToken = JWT.create()
+                .withSubject(UUID.randomUUID().toString())
```

Comment:
maybe we should swap the subject and the user id claim. the subject should identify to whom the token belongs